### PR TITLE
Allow specifying mac on macvlan link creation

### PR DESCRIPTION
**Pull request by maiqueb**
This probably should be added for other interface types, not only macvlan.

Welcome any feedback.

Fixes: #30

### CODE REVIEW

**Comment by maiqueb** [comment]
I still need to figure out how to add unit tests to this.

File: Cargo.toml [hunk 1]
```diff
@@ -28,6 +28,7 @@ netlink-proto = { default-features = false, version = "0.11" }
 nix = { version = "0.26.1", default-features = false, features = ["fs", "mount", "sched", "signal"] }
 tokio = { version = "1.0.1", features = ["rt"], optional = true}
 async-global-executor = { version = "2.0.2", optional = true }
+macaddr = "1.0"
```
Comment (maiqueb) [inline Cargo.toml:31]: not sure we want to consume this dependency.

**Review by cathay4t — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (maiqueb) [reply Cargo.toml:31]: Makes sense. 

As a follow up I can try to port (new implementation) the required functions into the project, thus removing the need for the dependency.

**Review by cathay4t — CHANGES_REQUESTED** [review]
CI failed.

I noticed we already have `LinkSetRequest::address()`. Can we introduce `LinkAddRequest::address()` to this PR. Otherwise this PR is purely for test and example code, does not make sense to have that PR title.

**Comment by maiqueb** [comment]
> CI failed.

Hm, locally I was using stable rust, and the ` wrap_comments` knob is only available in the nightly channel; thus, it didn't have the same output as CI. It should be fixed now @cathay4t . 

> 
> I noticed we already have `LinkSetRequest::address()`. Can we introduce `LinkAddRequest::address()` to this PR. Otherwise this PR is purely for test and example code, does not make sense to have that PR title.

Done. I've chosen to slightly refactor the `name` function as well (re-using the `append_nla` method) since that uses less code (and code reuse is good). Finally, I've chosen to make it public (just like in `LinkSetRequest`) to be able to use a builder like pattern (i.e. `link_handle.add().macvlan(...).name(<name>).address(<mac>)`).

I think this is ready for another review round @cathay4t .

Thanks for the help / reviews.

**Review by cathay4t — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**